src/bin/dual-iir.rs

```diff
@@ -519,7 +524,15 @@ mod app {
                     .inner_mut()]);
             });
 
-            c.local.usb_terminal.process().unwrap();
+            c.shared.settings.lock(|settings| {
+                c.local.usb_terminal.process(settings).unwrap();
+
+                c.shared.active_settings.lock(|active| {
+                    if active != &settings.dual_iir {
```

Comment:
This check is primarily the one I'm not fond of. We could update `usb_terminal::process()` to return a flag indicating if settings were modified as well to avoid this equality check.

Reply:
`process()` should return a bool whether the settings changed. Like miniconf's mqtt client does on `update()`.